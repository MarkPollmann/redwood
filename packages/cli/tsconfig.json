{
  "extends": "../../tsconfig.compilerOption.json",
  "compilerOptions": {
    "baseUrl": ".",
    "rootDir": "src",
    "outDir": "dist",
    "esModuleInterop": true
  },
  "include": ["src", "index.d.ts"],
}