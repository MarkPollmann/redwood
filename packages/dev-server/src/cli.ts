import yargs from 'yargs'
import {
  getSideConfig,
  getSidePaths,
  NodeTargetPaths,
} from '@redwoodjs/internal'

export const getArgsForSide = (
  side: string
): { port: number; host: string; functionsPath: string; watchPath: string } => {
  const { port, host } = getSideConfig(side)
  const paths = getSidePaths(side) as NodeTargetPaths

  return {
    port,
    host,
    functionsPath: paths.functions,
    watchPath: paths.base,
  }
}

const { side } = yargs.option('side', { default: 'api' }).argv

try {
  const { host, port, functionsPath, watchPath } = getArgsForSide(side)
} catch (e) {
  console.error(e.message)
  process.exit(1)
}
