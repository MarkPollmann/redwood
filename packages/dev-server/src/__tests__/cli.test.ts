import MockProject from '@redwoodjs/test-mocks'

import { getArgsForSide } from '../cli'

describe('cli', () => {
  it('getArgsForSide supplies the appropriate options to the dev server', () => {
    const mockfs = new MockProject()
    mockfs.mock()
    const args = getArgsForSide('api')
    mockfs.restore()
    expect(args).toMatchSnapshot()
  })
})
