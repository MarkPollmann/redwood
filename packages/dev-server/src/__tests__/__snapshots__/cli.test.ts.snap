// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli getArgsForSide supplies the appropriate options to the dev server 1`] = `
Object {
  "functionsPath": "/mock/project/api/src/functions",
  "host": "localhost",
  "port": 8911,
  "watchPath": "/mock/project/api",
}
`;
